feat: Update v12

dhuf opened the pull request.

dhuf committed:
feat: Update v12

dhuf committed:
feat: Make it run under v12

dhuf committed:
fix: Caching issue

dhuf committed:
chore: Remove code for v11

dhuf committed:
fix: Syntax

dhuf committed:
chore: Use rector

dhuf committed:
fix: phpunit config validation

dhuf committed:
fix: Tests

dhuf committed:
chore: Code refactoring

dhuf committed:
feat: add PHP 8.3

dhuf committed:
chore: Fix syntax

d-s-codappix commented:
@dhuf https://github.com/visol/ext-variables/pull/32 should fix the failing tests.

d-s-codappix committed:
Fix failing tests (#32)

* Add necessary TypoScript Configuration

The tests use `f:format.html` ViewHelper.
This will call the `ContentObjectRenderer::parseFunc`.

The default is to use `lib.parseFunc_RTE` as a reference.
But this didn't exist within our test setup, leading to errors.
We now ensure this path provides some configuration to pass the test.

* Follow PHPUnit naming schema for Test Cases

Our abstract TestCase wasn't named as one, leading to a PHPUnit warning.
We therefore adjust the naming to follow the convention and prevent any
warning.

* Ensure Database cache backend is used

As our tests use the database table to verify proper cache handling.
The testing framework replaced the defaults with `NullBackend` for
performance reasons.

* Solve PHPUnit Deprecation

DataProvider need to be static.

* Fix none working DataHandler tests

Pass necessary arguments to calls.
Migrate from unnecessary prophecy to native PHPUnit API.

dhuf committed:
feat: Check PHP 8.3
